lib/reduce_quda.cu

```diff
@@ -128,6 +128,36 @@ namespace quda {
   };
 
   /**
+     Return the L1 norm of x
+  */
+  __device__ double norm1_(const double2 &a) { return fabs(a.x) + fabs(a.y); }
+  __device__ float norm1_(const float2 &a) { return (double)fabs(a.x) + (double)fabs(a.y); }
```

Comment:
norm1_ returns float, so does the casting to double of the individual summands help in the precision of the reduction?

It is however consistent with the casting for norm2_